### PR TITLE
advanced spend builder

### PR DESCRIPTION
**Pull request by therealssj**
# Motivation
Allow the user to tweak his `coin hours`  and `unspents` in a transaction

# Info
- By default in a transaction, a user loses `75%` coin hours wherein `50%` is burned and `25%` is sent to destination addrs.
- The default strategy to select unspents priority is `Max skycoins` > `zero coin hours` > `non-zero coin hours` 

# Tasks
- [x] choose coin hours to be sent to destination addrs
- [x]  choose unspents to be used
- [x]  send from multiple addresses
- [x]  create `advancedCreateRawTx` command
- [x] Update changelog
- [x] Document use cases
- [ ] Tests 

# Use Cases
- This will allow users to send the desired number of coinhours to the receiver.
- Select unspent with lower coin hours to decrease transaction fees and hence preserving coinhours

### CODE REVIEW

**Comment by therealssj** [comment]
Could use some suggestions on the whole "different strategies to choose unspents" scenario

My main motivation behind this was the fact that we can decrease coin hour loss by using an unspent with fewer `coin hours` which has enough skycoins to satisfy the transaction but is not the one with the highest number of skycoins

File: cmd/cli/README.md [hunk 1]
```diff
@@ -135,6 +140,22 @@ $ skycoin-cli send -f $WALLET_PATH $recipient_address $amount
...
 Use `skycoin-cli send -h` to see the subcommand usage.
 
+### Advanced send
+
+```bash
+$ skycoin-cli advancedSend -s $coinhours_distribute -u $unspent_hash $recipient_address $amount
+```
+
+The above `advancedSend` command will send coins from your node's default wallet: `$HOME/.skycoin/wallets/skycoin_cli.wlt` if it has `$unspent_hash`. 
+It willt send `$coinhours_distribute` to `$recipient_address`
```
Comment (gz-c) [inline cmd/cli/README.md]: `willt` typo

Comment (gz-c) [reply cmd/cli/README.md]: The `$coinhours_distribute` parameter is not clear

**Comment by gz-c** [comment]
See #872, I wrote up an initial spec

**Comment by therealssj** [comment]
The example seems to be unclear but the command does let you use multiple unspents

**Comment by gz-c** [comment]
@therealssj should we close this?

**Comment by gz-c** [comment]
We'll redo this to use the `POST /wallet/transaction` API

**Pull request closed without merging**